src/compiler/utilities.ts

```diff
@@ -91,6 +91,11 @@ namespace ts {
             return false;
         }
 
+        if (sortBeforeComparison) {
+            arr1.sort();

```

Comment:
This also modifies the array in place. This will give the user unexpected behavior.